frontend/packages/knative-plugin/src/utils/get-knative-resources.ts

```diff
@@ -42,23 +42,29 @@ const getRevisions = (dc: K8sResourceKind, { revisions }): K8sResourceKind[] =>
   return revisionResource;
 };
 
-export const getKnativeServingRevisions = (dc: K8sResourceKind, props): KnativeItem => {
+export const getKnativeServingRevisions = (
+  dc: K8sResourceKind,
+  props,
+): KnativeItem | K8sResourceKind[] => {
   const revisions = getRevisions(dc, props);
-  return {
-    revisions,
-  };
+  const revsionsRes = revisions.length > 0 ? { revisions } : revisions;
```

Comment:
I'm a little confused. It looks like `getRevisions` always returns `K8sResourceKind[]`. So we're returning a `KnativeItem` is the array is not empty, but an empty array otherwise?

Reply:
in Overview, we are consuming this via extension utils and first check is if CRDs are present then  check is based on key as in knative specific resources "configurations" and in current code issue was even if knative resources were not present but utils returned 
```
{ configurations: [] }
```
https://github.com/openshift/console/blob/master/frontend/public/components/overview/index.tsx#L832

in case of normal deployments and as in sidebar it identifies configurations and knatuve operator is installed so treated it as knative resource.

 with added check in case of no configurations found will return [], as below so will not add any knative resources in overviewitems

```
overviewItems = { ...overviewItems, ...[] };
```